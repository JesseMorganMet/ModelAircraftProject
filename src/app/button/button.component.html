
<button (click)=thing(i)>{{d}}</button>
