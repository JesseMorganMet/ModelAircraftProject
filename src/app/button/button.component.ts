import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';

@Component({
  selector: 'app-button',
  templateUrl: './button.component.html',
  styleUrls: ['./button.component.scss']
})
export class ButtonComponent implements OnInit {
  @Input() d: any;
  @Input() i: any;
  @Output() example = new EventEmitter();

  constructor() { }

  ngOnInit() {
  }

  thing(i){
    this.example.emit(this.i);
  }

}
