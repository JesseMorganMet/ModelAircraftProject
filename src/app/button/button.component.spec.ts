import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtonComponent } from './button.component';

describe('ButtonComponent', () => {
  let component: ButtonComponent;
  let fixture: ComponentFixture<ButtonComponent>;
  let mockData = "placeName";
  let mockData2 = 1 ;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ButtonComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(ButtonComponent);
    component = fixture.componentInstance;
  });

  it('Should match snapshot', () => {
    component.d = mockData;
    component.i = mockData2;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
