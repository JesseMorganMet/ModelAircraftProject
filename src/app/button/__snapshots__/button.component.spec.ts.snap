// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Should match snapshot 1`] = `
<app-button
  __ngContext__="0"
  d={[Function String]}
  example={[Function EventEmitter_]}
  i={[Function Number]}
>
  <button>
    placeName
  </button>
</app-button>
`;
